### PR TITLE
Report fixes

### PR DESCRIPTION
**Pull request by saganshul**
This PR fixes many issues in new UI of report.

## Description
Changes:
* Previously, we make requests to server for every plugin. suppose if there are 136 plugins in report. Corresponding requests will be 136. This was hogging server due to large number of simultaneous requests. This PR eliminate that and sends the requests in parts of 30. So that there can max be 5 requests in case of 136. This eliminate hogging issue and also fasts the UI as we are doing the requests in parts.
* Previously, we make updated request(for data) for filtering to server anytime any filtering is requested. This PR eliminate server side filtering to client side filtering as we have all the data on load. So it is redundant to send the requests again. Now we just request the updated names of plugins after filtering which is very small request in term of response.

Above point makes UI superfast. As there is not server level delay after page load. Everything is dependent of client side after first load.

## Related Issue
#771 

## Reviewers
@delta24

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other

### CODE REVIEW

**Comment by saganshul** [comment]
@delta24 , can you please review this PR?

File: includes/src/Report/Accordians.js [hunk 1]
```diff
@@ -3,18 +3,32 @@ import Collapse from './Collapse'
...
+
         for (var i = 0; i < pluginDataList.length; i++) {
-            if (pluginDataList[i].user_rank != null && pluginDataList[i].user_rank != -1) {
-                if (pluginDataList[i].user_rank > maxUserRank) {
-                    maxUserRank = pluginDataList[i].user_rank;
-                }
-            } else if (pluginDataList[i].owtf_rank != null && pluginDataList[i].owtf_rank != -1) {
-                if (pluginDataList[i].owtf_rank > maxOWTFRank) {
-                    maxOWTFRank = pluginDataList[i].owtf_rank;
+            if ((selectedType.length === 0 || selectedType.indexOf(pluginDataList[i].plugin_type) !== -1) && (selectedGroup.length === 0 || selectedGroup.indexOf(pluginDataList[i].plugin_group) !== -1) && (selectedRank.length === 0 || selectedRank.indexOf(pluginDataList[i].user_rank) !== -1) && (selectedOwtfRank.length === 0 || selectedOwtfRank.indexOf(pluginDataList[i].owtf_rank) !== -1) && (selectedStatus.length === 0 || selectedStatus.indexOf(pluginDataList[i].status) !== -1)) {
```
Comment (delta24) [inline includes/src/Report/Accordians.js]: Be consistent when matching conditions:
`indexOf(pluginDataList[i].plugin_type) !== -1)`  and `pluginDataList[i].owtf_rank != -1`

File: includes/src/Report/Accordians.js [hunk 2]
```diff
@@ -66,29 +85,31 @@ class Accordians extends React.Component {
                                                 <div className="col-md-2">
                                                     <div className="btn-group btn-group-xs" role="group">
                                                         {pluginData[key].map(function(obj) {
-                                                            return (
-                                                                <button onClick={handlePluginBtnOnAccordian.bind(this, key, obj['plugin_type'])} key={key + obj['plugin_type'].split('_').join(' ')} className={(() => {
-                                                                    if (key in pluginData) {
-                                                                        if (testCaseMax == 0)
```
Comment (delta24) [inline includes/src/Report/Accordians.js]: Ditto.

File: includes/src/Report/Accordians.js [hunk 3]
```diff
@@ -66,29 +85,31 @@ class Accordians extends React.Component {
...
-                                                                    } else {
-                                                                        return "btn";
-                                                                    }
-                                                                })()} style={{
-                                                                    marginTop: "23px"
-                                                                }} type="button">{obj['plugin_type'].split('_').join(' ').charAt(0).toUpperCase() + obj['plugin_type'].split('_').join(' ').slice(1)}</button>
-                                                            );
+                                                            if ((selectedType.length === 0 || selectedType.indexOf(obj['plugin_type']) !== -1) && (selectedGroup.length === 0 || selectedGroup.indexOf(obj['plugin_group']) !== -1) && (selectedRank.length === 0 || selectedRank.indexOf(obj['user_rank']) !== -1) && (selectedOwtfRank.length === 0 || selectedOwtfRank.indexOf(obj['owtf_rank']) !== -1) && (selectedStatus.length === 0 || selectedStatus.indexOf(obj['status']) !== -1)) {
```
Comment (delta24) [inline includes/src/Report/Accordians.js]: Ditto.

File: includes/src/Report/Accordians.js [hunk 4]
```diff
@@ -66,29 +85,31 @@ class Accordians extends React.Component {
...
+                                                                return (
+                                                                    <button onClick={handlePluginBtnOnAccordian.bind(this, key, obj['plugin_type'])} key={key + obj['plugin_type'].split('_').join(' ')} className={(() => {
+                                                                        if (key in pluginData) {
+                                                                            if (testCaseMax == 0)
```
Comment (delta24) [inline includes/src/Report/Accordians.js]: Ditto.

**Comment by saganshul** [comment]
@delta24 PR updated. Please review. I have a done good level of optimisation in Report.(Most of unneccesary part of page do not re-render again and again). Also immutability also introduced in the project to maintain proper states update. I think you will not get any CPU high usage not atleast 100% IMO.

**Comment by saganshul** [comment]
@delta24 , can you please review this PR?

**Pull request merged**